packages/selected-network-controller/src/SelectedNetworkController.ts

```diff
@@ -332,22 +332,25 @@ export class SelectedNetworkController extends BaseController<
    * @returns The proxy and block tracker proxies.
    */
   getProviderAndBlockTracker(domain: Domain): NetworkProxy {
-    const networkClientId = this.getNetworkClientIdForDomain(domain);
     let networkProxy = this.#domainProxyMap.get(domain);
     if (networkProxy === undefined) {
       let networkClient;
-      if (networkClientId === undefined || !this.#useRequestQueuePreference) {
```

Comment:
`networkClientId` can't be undefined anymore (as of https://github.com/MetaMask/core/pull/3908/files#diff-5e705302d595033f24cac5f20d14ecf307709fd61c46a6d01bf9519bf5b91d86R204)